tests/assembly/simd-bitmask.rs

```diff
@@ -0,0 +1,149 @@
+//@ revisions: x86 x86-avx2 x86-avx512 aarch64
+//@ [x86] compile-flags: --target=x86_64-unknown-linux-gnu -C llvm-args=-x86-asm-syntax=intel
+//@ [x86] needs-llvm-components: x86
+//@ [x86-avx2] compile-flags: --target=x86_64-unknown-linux-gnu -C llvm-args=-x86-asm-syntax=intel
+//@ [x86-avx2] compile-flags: -C target-feature=+avx2
+//@ [x86-avx2] needs-llvm-components: x86
+//@ [x86-avx512] compile-flags: --target=x86_64-unknown-linux-gnu -C llvm-args=-x86-asm-syntax=intel
+//@ [x86-avx512] compile-flags: -C target-feature=+avx512f,+avx512vl,+avx512bw,+avx512dq
+//@ [x86-avx512] needs-llvm-components: x86
+//@ [aarch64] compile-flags: --target=aarch64-unknown-linux-gnu
+//@ [aarch64] needs-llvm-components: aarch64
+//@ [aarch64] min-llvm-version: 18.0
+//@ assembly-output: emit-asm
+//@ compile-flags: --crate-type=lib -O
+
+#![feature(no_core, lang_items, repr_simd, intrinsics)]
+#![no_core]
+#![allow(non_camel_case_types)]
+
+// Because we don't have core yet.
+#[lang = "sized"]
+pub trait Sized {}
+
+#[lang = "copy"]
+trait Copy {}
+
+#[repr(simd)]
+pub struct m8x16([i8; 16]);
+
+#[repr(simd)]
+pub struct m8x64([i8; 64]);
+
+#[repr(simd)]
+pub struct m32x4([i32; 4]);
+
+#[repr(simd)]
+pub struct m64x2([i64; 2]);
+
+#[repr(simd)]
+pub struct m64x4([i64; 4]);
+
+extern "rust-intrinsic" {
+    fn simd_bitmask<V, B>(mask: V) -> B;
+}
+
+// CHECK-LABEL: bitmask_m8x16
+#[no_mangle]
+pub unsafe extern "C" fn bitmask_m8x16(mask: m8x16) -> u16 {
+    // The simd_bitmask intrinsic already uses the most significant bit, so no shift is necessary.
+    // Note that x86 has no byte shift, llvm uses a word shift to move the least significant bit
+    // of each byte into the right position.
+    //
+    // x86-NOT: psllw
+    // x86: movmskb eax, xmm0
+    //
+    // x86-avx2-NOT: vpsllw
+    // x86-avx2: vpmovmskb eax, xmm0
+    //
+    // x86-avx512-NOT: vpsllw xmm0
+    // x86-avx512: vpmovmskb eax, xmm0
+    //
+    // aarch64: adrp
+    // aarch64-NEXT: cmlt
+    // aarch64-NEXT: ldr
+    // aarch64-NEXT: and
+    // aarch64-NEXT: ext
+    // aarch64-NEXT: zip1
+    // aarch64-NEXT: addv
+    // aarch64-NEXT: fmov
```

Comment:
This looks ok. Full listing including the constants:
```asm
           5: .LCPI0_0: 
           6:  .byte 1 
           7:  .byte 2 
           8:  .byte 4 
           9:  .byte 8 
          10:  .byte 16 
          11:  .byte 32 
          12:  .byte 64 
          13:  .byte 128 
          14:  .byte 1 
          15:  .byte 2 
          16:  .byte 4 
          17:  .byte 8 
          18:  .byte 16 
          19:  .byte 32 
          20:  .byte 64 
          21:  .byte 128 
          22:  .section .text.bitmask_m8x16,"ax",@progbits 
          23:  .globl bitmask_m8x16 
          24:  .p2align 2 
          25:  .type bitmask_m8x16,@function 
          26: bitmask_m8x16: 
          27:  .cfi_startproc 
          28:  adrp x8, .LCPI0_0 
          29:  cmlt v0.16b, v0.16b, #0 
          30:  ldr q1, [x8, :lo12:.LCPI0_0] 
          31:  and v0.16b, v0.16b, v1.16b 
          32:  ext v1.16b, v0.16b, v0.16b, #8 
          33:  zip1 v0.16b, v0.16b, v1.16b 
          34:  addv h0, v0.8h 
          35:  fmov w0, s0 
          36:  ret 
```

Reply:
Nice.